### PR TITLE
Improve docs to avoid confusion related to the core git-sync feature

### PR DESCRIPTION
**Pull request by danielhoherd**
## Description

Improve docs to avoid confusion related to the core git-sync feature, and other small improvements. This is in response to the issue in this thread:

- <https://astronomer.slack.com/archives/C02NZELT2/p1671464975671469>
- <https://apache-airflow.slack.com/archives/C03T0AVNA6A/p1671450081264109>

## Related Issues

N/A

## Testing

Only docs changes, no testing is needed. I did set up a kind cluster and test these instructions out while changing them.

Because these are only docs changes, the only CI step that needs to pass is run_pre_commit

## Merging

Merge everywhere git-sync-relay is supported.

### CODE REVIEW

**Review by jedcunningham — APPROVED** [review]

File: docs/git-sync-relay.md [hunk 1]
```diff
@@ -36,18 +36,18 @@ Create a `values.yaml` file with contents similar to the following:
...
     wait: 60 # seconds between synchronizations with upstream git repo
     subPath: dags # if your dags dir is not the repo root, specify the path relative to the repo root
-    sshPrivateKeySecretName: theGitSshPrivateKeyName # The name of a secret that holds the private key.
+    sshPrivateKeySecretName: git-ssh-private-key # This is the secret we created earlier. This is not requred with https git remotes.
 ```
 
 ### Install airflow
 
-Then install airflow like you normally would. Assuming your cwd is the root dir or the astronomer/airflow-chart repo:
+You must also follow the git-sync configuration setup that you would normally follow to install the apache airflow chart. The one customization you must make is to use the git repo served by the git-sync-relay's git-daemon pod as your git remote. That URL is `git://airflow-git-sync-relay.${NAMESPACE}.svc.cluster.local./git` where `${NAMESPACE}` is the airflow installation namespace. Once you have those values set up in your values.yaml file, you can install the astronomer airflow-chart:
```
Comment (jedcunningham) [inline docs/git-sync-relay.md:49]: An example might be helpful. Is there any auth on the relay?

Comment (danielhoherd) [reply docs/git-sync-relay.md:49]: git-daemon serves using git protocol, which has no auth and is quite performant. security is maintained by keeping the git-daemon restricted to the local namespace.

What kind of example are you thinking of?

Comment (jedcunningham) [reply docs/git-sync-relay.md:49]: Cool. I meant an example values, something like this maybe:

```
airflow:
  dags:
    gitSync:
      enable: true
      repo: git://airflow-git-sync-relay/git`
```

And to make it simpler, just using the service name should be okay right?

```suggestion
You must also follow the git-sync configuration setup that you would normally follow to install the apache airflow chart. The one customization you must make is to use the git repo served by the git-sync-relay's git-daemon pod as your git remote. That URL is `git://airflow-git-sync-relay/git`. Once you have those values set up in your values.yaml file, you can install the astronomer airflow-chart:
```

Comment (danielhoherd) [reply docs/git-sync-relay.md]: @jedcunningham can you check the current version and see if it's suitable?

File: docs/git-sync-relay.md [hunk 2]
```diff
@@ -36,18 +36,28 @@ Create a `values.yaml` file with contents similar to the following:
...
+
+airflow:
+  dags:
+    gitSync:
+      enabled: True
+      repo: git://airflow-git-sync-relay.${NAMESPACE}.svc.cluster.local./git
+      branch: main
+  dagDeployment:
+    repositoryUrl: https://github.com/astronomer/2-4-example-dags
+    type: git_sync
```
Comment (jedcunningham) [inline docs/git-sync-relay.md:54]: This must be in the wrong place, that isn't in the OSS chart.

File: docs/git-sync-relay.md (hunk 1 above)
Comment (jedcunningham) [reply docs/git-sync-relay.md]: Sorry, missed this! Looks good.

File: docs/git-sync-relay.md (hunk 2 above)
Comment (danielhoherd) [reply docs/git-sync-relay.md:54]: That's bizarre. I did a deploy in dev and inspected the values and that's the structure I saw, so I configured it and it did work. I'm going to have to go back and look again and figure out wtf is going on because I don't see dagDeployment in either this chart or the OSS chart.

Comment (danielhoherd) [reply docs/git-sync-relay.md:54]: Oh, my guess is that houston is storing state in the helm values. I'm doing another test now to see what the bare minimum instructions we need to include in doc to get it working are.

Comment (danielhoherd) [reply docs/git-sync-relay.md:54]: I don't know what I was doing wrong before that was causing a crashloop, but when I try your example today it's working so yay.

Comment (jedcunningham) [reply docs/git-sync-relay.md:54]: The best bugfix: time 🙃
